tasks/omnibus.py

```diff
@@ -404,7 +404,23 @@ def rpath_edit(ctx, install_path, target_rpath_dd_folder, platform="linux"):
         else:
             if file_type != "application/x-mach-binary":
                 continue
-            binary_rpath = ctx.run(f'otool -l {file} | grep -A 2 "RPATH"', warn=True, hide=True).stdout
+            errors = ctx.run(f'otool -l {file} > tmpotoolfile', warn=True, hide=True).stderr
+            if errors:
+                continue
+            binary_rpath = ctx.run('cat tmpotoolfile | grep -A 2 "RPATH"', warn=True, hide=True).stdout
+            dylib_paths = ctx.run(
+                r'cat tmpotoolfile | grep -A 2 "LC_LOAD_DYLIB\|LC_ID_DYLIB"', warn=True, hide=True
+            ).stdout
+            if install_path in dylib_paths:
+                for otool_line in dylib_paths.splitlines():
+                    if "name" not in otool_line:
+                        continue
+                    dylib_path = otool_line.strip().split(" ")[1]
+                    if install_path not in dylib_path:
+                        continue
+                    new_dylib_path = dylib_path.replace(install_path, "@rpath")
+                    # Edit LC_LOAD_DYLIB paths relative to rpath
+                    ctx.run(f"install_name_tool -change {dylib_path} {new_dylib_path} {file}")
```

Comment:
`install_name_tool` seems to accept [multiple](https://keith.github.io/xcode-man-pages/install_name_tool.1.html#change) `-change` arguments on the same invokation. Would it be worth trying to group all of these changes?

Reply:
😮 didn't notice it was possible will try, if yes then definitely worth 🚀

Reply:
as long we don't reach the shell argument limit

Reply:
> as long we don't reach the shell argument limit

I would be surprised if we did, that would mean a single binary depending on hundreds of libraries.

Reply:
Last commit reduce the time ~30 sec n oneed to compact further lets play safe and keep it as is